gax/src/test/java/com/google/api/gax/rpc/ClientContextTest.java

```diff
@@ -120,6 +120,16 @@ public TransportChannelProvider withHeaders(Map<String, String> headers) {
     }
 
     @Override
+    public boolean needsEndpoint() {
+      return false;
+    }
+
+    @Override
+    public TransportChannelProvider withEndpoint(String endpoint) {
+      return this;
```

Comment:
This should throw?

Reply:
This is a FakeTransportProvider, so it doesn't matter much - I was being consistent with the rest of this class, which is permissive.